src/filters/write_generic.c

```diff
@@ -936,7 +936,11 @@ static GF_Err writegen_push_ttml(GF_GenDumpCtx *ctx, char *data, u32 data_size,
 	}
 
 	if (!ctx->ttml_root) {
-		ctx->ttml_root = gf_xml_dom_detach_root(dom);
+		root_global = gf_xml_dom_detach_root(dom);
+		if (root_global) {
+			if (gf_list_count(root_global->content) > 0) ctx->ttml_root = root_global;
+			else gf_free(root_global);
```

Comment:
this should be gf_xml_dom_node_del, not gf_free. Apart from that looks goot to me

Reply:
Fixed